src/target/riscv/riscv-013.c

```diff
@@ -1712,8 +1720,12 @@ static int examine(struct target *target)
 	}
 
 	riscv013_info_t *info = get_info(target);
-	/* TODO: This won't be true if there are multiple DMs. */
-	info->index = target->coreid;
+	/* As info->index is the index of each hart on the DM (Debug Module),
+	 * for each DM these must start at 0 and be contiguous.
+	 * And target->coreid is target->coreid is globally unique in OpenOCD.
+	 * So for multiple DMs, info->index and target->coreid can't be the same.
+	 * Consider different DM have different dbgbase, let coreid = index + dbgbase */
+	info->index = target->coreid - target->dbgbase;
```

Comment:
should I add  `if (target->coreid >= target->dbgbase)` for ` info->index = target->coreid - target->dbgbase;`